diff --git a/dlib/CMakeLists.txt b/dlib/CMakeLists.txt
index 83230606f4..26457c85c3 100644
--- a/dlib/CMakeLists.txt
+++ b/dlib/CMakeLists.txt
@@ -694,19 +694,9 @@ if (NOT TARGET dlib)
                # make sure cuda is really working by doing a test compile
                message(STATUS "Building a CUDA test project to see if your compiler is compatible with CUDA...")
 
-               set(CUDA_TEST_CMAKE_FLAGS 
-                  "-DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}"
-                  "-DCMAKE_INCLUDE_PATH=${CMAKE_INCLUDE_PATH}"
-                  "-DCMAKE_LIBRARY_PATH=${CMAKE_LIBRARY_PATH}")
-
-               if (NOT MSVC) # see https://github.com/davisking/dlib/issues/363
-                  list(APPEND CUDA_TEST_CMAKE_FLAGS "-DCUDA_HOST_COMPILER=${CUDA_HOST_COMPILER}")
-               endif()
-
                try_compile(cuda_test_compile_worked 
                   ${PROJECT_BINARY_DIR}/cuda_test_build 
                   ${PROJECT_SOURCE_DIR}/cmake_utils/test_for_cuda cuda_test
-                  CMAKE_FLAGS ${CUDA_TEST_CMAKE_FLAGS}
                   OUTPUT_VARIABLE try_compile_output_message
                   )
                if (NOT cuda_test_compile_worked)
@@ -722,7 +712,6 @@ if (NOT TARGET dlib)
                   try_compile(cudnn_test_compile_worked 
                      ${PROJECT_BINARY_DIR}/cudnn_test_build 
                      ${PROJECT_SOURCE_DIR}/cmake_utils/test_for_cudnn cudnn_test
-                     CMAKE_FLAGS ${CUDA_TEST_CMAKE_FLAGS}
                      OUTPUT_VARIABLE try_compile_output_message
                      )
                   if (NOT cudnn_test_compile_worked)
